Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Internal/DriverMode.hs

```diff
@@ -40,5 +45,12 @@ setDriverMode apiKey req = do
   locationTrackingServiceKey <- asks (.locationTrackingServiceKey)
   unless (apiKey == Just locationTrackingServiceKey) $ do
     throwError $ InvalidRequest "Invalid API key"
-  void $ QDriverInformation.updateActivity isActive (Just mode) driverId
+  void $ do
+    QDriverInformation.updateActivity isActive (Just mode) driverId
+    driver <- QPerson.findById driverId >>= fromMaybeM (PersonDoesNotExist driverId.getId)
+    transporterConfig <-
+      SCTC.findByMerchantOpCityId driver.merchantOperatingCityId Nothing
+        >>= fromMaybeM (TransporterConfigNotFound driver.merchantOperatingCityId.getId)
+    driverInfo <- SQDI.findByPrimaryKey driverId >>= fromMaybeM DriverInfoNotFound
+    processingChangeOnline (driverId, driver.merchantId, driver.merchantOperatingCityId) transporterConfig.timeDiffFromUtc driverInfo $ Just mode
```

Comment:
you can use fork and redis lock everywhere, or move it inside of `processingChangeOnline` implementation